### PR TITLE
`asm-ingress-gateway`: add hpa/role + simplify deployment/service deployment

### PR DESCRIPTION
**Pull request by mathieu-benoit**
`asm-ingress-gateway`: add `hpa`/`role` + simplify `deployment`/`service` deployment.

Reviewed the region tags, not used in docs.

### CODE REVIEW

File: docs/ingress-gateway-asm-manifests/base/deployment-service.yaml
Comment (xtineskim) [inline docs/ingress-gateway-asm-manifests/base/deployment-service.yaml:54]: thinking back to the discussion in asm packages https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/pull/1244

Comment (mathieu-benoit) [reply docs/ingress-gateway-asm-manifests/base/deployment-service.yaml:54]: Yeah, good discussion in there, but it was more about migrating existing customer leveraging this manifests (changes on the `Service` will cause downtime). In the case of this PR here, I'm not changing that, both `Deployment` and `Service` remain the same, I just moved them under the same file now.
This PR also adds HPA, Role and RoleBinding.
Does that clarify? Let me know if you have any other feedback/question. Thanks!

Comment (xtineskim) [reply docs/ingress-gateway-asm-manifests/base/deployment-service.yaml:54]: makes sense, thanks 👍

File: docs/ingress-gateway-asm-manifests/base/deployment-service.yaml
Comment (NimJay) [inline docs/ingress-gateway-asm-manifests/base/deployment-service.yaml:130]: **Thought:**
I wonder if we should do the same for https://github.com/GoogleCloudPlatform/anthos-service-mesh-samples/tree/main/docs/shared/asm-ingress-gateway?
No action required (inside this PR).

**Review by xtineskim — APPROVED** [review]

**Review by NimJay — APPROVED** [review]
Looks good to me.
I don't believe we need to update any docs, right?
Reason: I believe this is only used by [Strengthen your app's security with Anthos Service Mesh and Anthos Config Management](https://cloud.google.com/anthos-config-management/docs/tutorials/strengthen-app-security) and these samples are used indirectly.

Thanks, Mathieu, for these improvements.
I would wait for Christine's approval (inside her comment) before merging.
LGTM.

File: docs/ingress-gateway-asm-manifests/base/deployment-service.yaml
Comment (mathieu-benoit) [reply docs/ingress-gateway-asm-manifests/base/deployment-service.yaml:130]: Yep good point, we let @xtineskim decide, the context of that is that as best practice we should show/have this. We discussed that with a CEs based on their customer feedback on that guide: https://cloud.google.com/architecture/exposing-service-mesh-apps-through-gke-ingress, so took the opportunity to address that here too.

**Comment by mathieu-benoit** [comment]
Yep, only that docs for now, and impact measured with the Kustomize overlay/package around that too.

Thanks you two for your review!

**Pull request merged**
